modules/javafx.graphics/src/main/java/javafx/scene/layout/AnchorPane.java

```diff
@@ -355,15 +350,15 @@ private double computeChildHeight(Node child, Double topAnchor, Double bottomAnc
             }
 
             if (leftAnchor != null) {
-                x = insets.getLeft() + leftAnchor;
+                x = snappedLeftInset() + leftAnchor;
             } else if (rightAnchor != null) {
-                x = getWidth() - insets.getRight() - rightAnchor - w;
+                x = getWidth() - snappedRightInset() - rightAnchor - w;
             }
 
             if (topAnchor != null) {
-                y = insets.getTop() + topAnchor;
+                y = snappedTopInset() + topAnchor;
             } else if (bottomAnchor != null) {
-                y = getHeight() - insets.getBottom() - bottomAnchor - h;
+                y = getHeight() - snappedBottomInset() - bottomAnchor - h;
             }
 
             child.resizeRelocate(x, y, w, h);
```

Comment:
snapping should be done at this moment, I think, something like this:
```
child.resizeRelocate(
  snapPositionX(x),
  snapPositionY(y),
  snapSizeX(w),
  snapSizeY(h));
```
and keep the original (unsnapped) code prior to this.

Reply:
It depends (see earlier discussion), but if we are going to snap the intermediate results, we need to be sure we are doing it consistently -- adding snapped insets to an unsnapped value is very likely the wrong thing to do.

Reply:
not only for consistency, but we should also understand whether it makes sense or not.

for example, we might snap intermediate results such as guide lines, if that makes sense (e.g. for alignment), but most importantly, we must honor the snapToPixel setting for the end result.  If it's true, then all child nodes must have integer coordinates and sizes.

Would you agree?